### DISCUSSION TITLE
Obfuscate my files and push to repository

### DISCUSSION THREAD

**Discussion by shiva-sandupatla** [community/community#26046] [Actions]
<p><strong>Hi, all.</strong></p>
<p><strong>Hope you are going.</strong></p>
<p><strong>is it possible to obfuscate my files when i pushed to my repository and after obfuscated it should be pushed to another repository.</strong></p>
<p><strong>Is it possible using actions or is their any way to do like that.</strong></p>
<p><strong>Hoping with a positive response.</strong></p>
<p><strong>Regards, shiva</strong></p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/shivani-designer">@shivani-designer</a>,</p>
<p>Do you mean that after pushing some modified or added files to a repository, then use workflow to automatically sync the new changes to another repository?<br>
If so, here is an example workflow as reference:</p>
<pre><code class="lang-yaml"># Suppose you will manually push new commits to GitHub repository 'Repos-01'
# and want to sync code to another GitHub repository 'Repos-02'.
# You can setup this workflow in repository 'Repos-01'.

name: Sync code

on: push
jobs:
  sync_code:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout code from Repos-01
        uses: actions/checkout@v2
        with:
          persist-credentials: false
          path: Repos-01

      - name: Remove remote
        working-directory: Repos-01
        run: find . -type f | grep -i "\.git" | xargs rm

      - name: Checkout code from Repos-02
        uses: actions/checkout@v2
        with:
          repository: 'owner/Repos-02'
          token: ${{ secrets.MY_GITHUB_PAT }}
          path: Repos-02

      - name: Copy files
        run: cp -r Repos-01/* Repos-02

      - name: Push to Repos-02
        working-directory: Repos-02
        run: |
          git config --global user.name "user name"
          git config --global user.email "user email"
          git add -A
          git commit -m "commit message"
          git push
</code></pre>

**Comment by shiva-sandupatla** [comment]
<p>Thanks for the reply <a class="mention" href="/brightran">@brightran</a> .</p>
<p>If you don’t mind can you tell, How can i obfuscate those files which are containing .js extension.</p>
<p>And rather than pushing to another repository i want to push to another branch is it possible?</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/shivani-designer">@shivani-designer</a>,</p>
<blockquote>
<p>How can i obfuscate those files which are containing .js extension.</p>
</blockquote>
<p>What do you mean about “<strong>obfuscate</strong> files” ? Can you give an example to explain the actual behavior you want.</p>
<blockquote>
<p>And rather than pushing to another repository i want to push to another branch is it possible?</p>
</blockquote>
<p>You can try to use the <a href="https://git-scm.com/docs/git-merge" rel="nofollow noopener"><strong>git-merge</strong></a> command to merge the new commits from a branch into another branch in the same repository.</p>

**Comment by shiva-sandupatla** [comment]
<p><a class="mention" href="/brightran">@brightran</a>,</p>
<aside class="quote group-github-partners" data-username="brightran" data-post="4" data-topic="128522">
<div class="title">
<div class="quote-controls"></div>
 brightran:</div>
<blockquote>
<p>What do you mean about “ <strong>obfuscate</strong> files” ?</p>
</blockquote>
</aside>
<p>Their is a tool called <a href="https://obfuscator.io" rel="nofollow noopener">https://obfuscator.io</a> for obfuscate javascript files.</p>
<p>For example:</p>
<p>if you have javascript file as following code:</p>
<p><code>function hi(){</code><br>
<code>console.log('hellow world');</code><br>
<code>} hi();</code></p>
<p>The above code will be converted to the file code</p>
<p><code>var _0xb5c3=['\x5b\x5e\x20\x5d\x2b\x29\x2b\x29\x2b\x5b','\x6c\x6f\x67','\x74\x65\x73\x74','\x5e\x20\x5d\x7d','\x61\x70\x70\x6c\x79','\x72\x65\x74\x75\x72\x6e\x20\x2f\x22\x20','\x5e\x28\x5b\x5e\x20\x5d\x2b\x28\x20\x2b','\x48\x65\x6c\x6c\x6f\x20\x57\x6f\x72\x6c'];(function(_0x1cc120,_0x44291b){var _0x11419d=function(_0x5c5c23){while(--_0x5c5c23){_0x1cc120['push'](_0x1cc120['shift']());}},_0x213fd4=function(){var _0x7a6eea={'data':{'key':'cookie','value':'timeout'},'setCookie':function(_0xf55458,_0x3e7a67,_0x2a9eeb,_0x3b09ba){_0x3b09ba=_0x3b09ba||{};var _0x2c14e1=_0x3e7a67+'='+_0x2a9eeb,_0x46ff37=0x1f*0x14+-0x1*0x1dc4+0xe*0x1f4;for(var _0x35fd98=0x131a+-0x1*0x259d+0x1283,_0x5b3236=_0xf55458['length'];_0x35fd98&lt;_0x5b3236;_0x35fd98++){var _0x38c0bc=_0xf55458[_0x35fd98];_0x2c14e1+=';\x20'+_0x38c0bc;var _0x45d46d=_0xf55458[_0x38c0bc];_0xf55458['push'](_0x45d46d),_0x5b3236=_0xf55458['length'],_0x45d46d!==!![]&amp;&amp;(_0x2c14e1+='='+_0x45d46d);}_0x3b09ba['cookie']=_0x2c14e1;},'removeCookie':function(){return'dev';},'getCookie':function(_0x9f40d9,_0x58eea9){_0x9f40d9=_0x9f40d9||function(_0x55c652){return _0x55c652;};var _0x320814=_0x9f40d9(new RegExp('(?:^|;\x20)'+_0x58eea9['replace'](/([.$?*|{}()[]\/+^])/g,'$1')+'=([^;]*)')),_0x145a7b=function(_0x1db3c3,_0x444fe5){_0x1db3c3(++_0x444fe5);};return _0x145a7b(_0x11419d,_0x44291b),_0x320814?decodeURIComponent(_0x320814[0x1d4d*0x1+-0x444+-0x10b*0x18]):undefined;}},_0x3e926d=function(){var _0x58bbc7=new RegExp('\x5cw+\x20*\x5c(\x5c)\x20*{\x5cw+\x20*[\x27|\x22].+[\x27|\x22];?\x20*}');return _0x58bbc7['test'](_0x7a6eea['removeCookie']['toString']());};_0x7a6eea['updateCookie']=_0x3e926d;var _0xa38c04='';var _0x2f81a1=_0x7a6eea['updateCookie']();if(!_0x2f81a1)_0x7a6eea['setCookie'](['*'],'counter',-0x1*0xf55+0x14f0+-0x3*0x1de);else _0x2f81a1?_0xa38c04=_0x7a6eea['getCookie'](null,'counter'):_0x7a6eea['removeCookie']();};_0x213fd4();}(_0xb5c3,0x59*0x67+0x549*-0x1+-0x1d4e));var _0x4841=function(_0x4700dc,_0x1c8738){_0x4700dc=_0x4700dc-(0x1f*0x14+-0x1*0x1dc4+0xe*0x1f4);var _0x9d5f3=_0xb5c3[_0x4700dc];return _0x9d5f3;};function hi(){var _0x2c789a=function(){var _0x38d1fb=!![];return function(_0x2334e5,_0x5b6484){var _0x293cc9=_0x38d1fb?function(){if(_0x5b6484){var _0x48adad=_0x5b6484[_0x4841('\x30\x78\x34')](_0x2334e5,arguments);return _0x5b6484=null,_0x48adad;}}:function(){};return _0x38d1fb=![],_0x293cc9;};}(),_0x18ffba=_0x2c789a(this,function(){var _0x41ff5c=function(){var _0x564f12=_0x41ff5c['\x63\x6f\x6e\x73\x74\x72\x75\x63\x74\x6f'+'\x72'](_0x4841('\x30\x78\x35')+'\x2b\x20\x74\x68\x69\x73\x20\x2b\x20\x22'+'\x2f')()['\x63\x6f\x6e\x73\x74\x72\x75\x63\x74\x6f'+'\x72'](_0x4841('\x30\x78\x36')+_0x4841('\x30\x78\x30')+_0x4841('\x30\x78\x33'));return!_0x564f12[_0x4841('\x30\x78\x32')](_0x18ffba);};return _0x41ff5c();});_0x18ffba(),console[_0x4841('\x30\x78\x31')](_0x4841('\x30\x78\x37')+'\x64\x21');}hi();</code></p>
<p>So that anyone can’t read it and only browser can read it and debug it.</p>
<p>And they (<a href="https://obfuscator.io" rel="nofollow noopener">https://obfuscator.io</a>) provided a way to obfuscate code:</p>
<ul>
<li>using <strong>NPM</strong>
</li>
</ul>
<blockquote>
<p>Iam new to actions i don’t know how it can be wrote</p>
</blockquote>
<p>So want my main aim is</p>
<ul>
<li>I have repo named repo1 of two branches namly 1. Master 2. code</li>
<li>I will push my code to “code” branch. So when i push to this branch.</li>
<li>Only files with .js should be obfuscate using <strong>NPM</strong>
</li>
<li>And push this obfuscated file to <strong>MASTER</strong>  branch.<br>
Using actions.</li>
</ul>
<p>If you don’t mind can you code for my aim.</p>
<p>Hoping with a positive response :innocent:</p>
<p>Regards, shivani</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/shivani-designer">@shivani-designer</a>,</p>
<p>In your repository, you can try to setup a workflow on the <strong>code</strong> branch like as this:</p>
<pre><code class="lang-yaml">name: Obfuscate JavaScript code

on: push
jobs:
  obfuscate_code:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout code from code branch
        uses: actions/checkout@v2
        with:
          ref: code
          path: b_code

      - name: Remove remote
        working-directory: b_code
        run: find . -type f | grep -i "\.git" | xargs rm

      # In this step you generally need to run some related command lines
      # of the JavaScript Obfuscator Tool to obfuscate the JavaScript code.
      # see 'https://github.com/javascript-obfuscator/javascript-obfuscator#cli-usage'
      - name: Obfuscate JavaScript code
        working-directory: b_code
        run: |
          --- some command lines to obfuscate the JavaScript code ---

      - name: Checkout code from master branch
        uses: actions/checkout@v2
        with:
          ref: master
          path: b_master

      # Copy files from b_code into b_master, the new obfuscated JavaScript files
      # will be copied to overwrite the old obfuscated JavaScript files in b_master.
      - name: Copy files
        run: cp -r b_code/* b_master

      - name: Push to remote master branch
        working-directory: b_master
        run: |
          git config --global user.name "user name"
          git config --global user.email "user email"
          git add -A
          git commit -m "Obfuscate JavaScript code"
          git push
</code></pre>

**Comment by shiva-sandupatla** [comment]
<p><a class="mention" href="/brightran">@brightran</a>,</p>
<p>I setup workflow  as you said above</p>
<pre><code class="lang-auto">name: Obfuscate JavaScript code

on: push


jobs:
  obfuscate_code:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout code from code branch
        uses: actions/checkout@v2
        with:
          ref: code
          path: b_code

      - name: Remove remote
        working-directory: b_code
        run: find . -type f | grep -i "\.git" | xargs rm

      # In this step you generally need to run some related command lines
      # of the JavaScript Obfuscator Tool to obfuscate the JavaScript code.
      # see 'https://github.com/javascript-obfuscator/javascript-obfuscator#cli-usage'
      - name: Obfuscate JavaScript code
        working-directory: b_code
        run: |
      
          javascript-obfuscator undex.js --output undex.js [options]
        
          

      - name: Checkout code from master branch
        uses: actions/checkout@v2
        with:
          ref: master
          path: b_master

      # Copy files from b_code into b_master, the new obfuscated JavaScript files
      # will be copied to overwrite the old obfuscated JavaScript files in b_master.
      - name: Copy files
        run: cp -r b_code/* b_master

      - name: Push to remote master branch
        working-directory: b_master
        run: |
          git config --global user.name "the-bug-finder"
          git config --global user.email "sandupatlashiva2@gmail.com"
          git add -A
          git commit -m "Obfuscate JavaScript code"
          git push
</code></pre>
<p>and i got an error in it.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/3367266da7a041c78ae26655225a9230e061c531.png" data-download-href="/uploads/short-url/7kJlDSVHEMwvgtZuraSop4aHuI9.png?dl=1" title="Screenshot (222)" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/47274553/181094967-c6cdf386-b137-47a8-ab4e-7bbc4c740a28.png" alt="Screenshot (222)" data-base62-sha1="7kJlDSVHEMwvgtZuraSop4aHuI9" width="690" height="387" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/3367266da7a041c78ae26655225a9230e061c531_2_690x387.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/3367266da7a041c78ae26655225a9230e061c531_2_1035x580.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/3/3367266da7a041c78ae26655225a9230e061c531.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/3367266da7a041c78ae26655225a9230e061c531_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot (222)</span><span class="informations">1366×768 69.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>if you don’t mind can i contact you personally to solve my problem.</p>

**Comment by airtower-luna** [comment]
<p>The “command not found” error occurs because there is no step in your workflow that installs the <code>javascript-obfuscator</code> command, and it’s not part of the <a href="https://github.com/actions/virtual-environments/blob/4865e6e3947bfb2cc7eb91eb66a99c0a46121f10/images/linux/Ubuntu1804-README.md">software installed by default on <code>ubuntu-latest</code></a> either. :wink:</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/shivani-designer">@shivani-designer</a>,</p>
<p>As <a class="mention" href="/airtower-luna">@airtower-luna</a> mentioned, the JavaScript obfuscator is not the default software installed on GitHub-hosted runners. You need to install this tool before using it.</p>
<p>In the <a href="https://github.com/javascript-obfuscator/javascript-obfuscator#javascript-obfuscator" rel="nofollow noopener"><strong>README</strong></a> of the the JavaScript obfuscator, it has shared the main commands to install this tool (see <a href="https://github.com/javascript-obfuscator/javascript-obfuscator#installation" rel="nofollow noopener">here</a>).<br>
In your workflow, you can reference the installation steps in the <strong>README</strong> to set a step for installing the JavaScript obfuscator on the runner.</p>
